homeworks/07-js-basics-2/index.js

```diff
@@ -0,0 +1,196 @@
+// 1 https://www.codewars.com/kata/5715eaedb436cf5606000381
+
+function positiveSum(arr) {
+  return arr.reduce((acc, el) => {
+    if (el > 0) {
+      return acc + el;
+    } else {
+      return acc;
+    }
+  }, 0);
+}
+
+// 2 https://www.codewars.com/kata/5a3e1319b6486ac96f000049
+
+function pairs(ar) {
+  let count = 0;
+
+  for (let i = 0; i < ar.length; i = i + 2) {
+    if (ar[i] + 1 === ar[i + 1] || ar[i] - 1 === ar[i + 1]) {
+      count++;
+    }
+  }
+
+  return count;
+}
+
+// 3 https://www.codewars.com/kata/5aba780a6a176b029800041c
+
+function maxMultiple(divisor, bound) {
+  let result = 0;
+
+  let i = 1;
+  while (i <= bound) {
+    if (i % divisor === 0) {
+      result = i;
+    }
+    i++;
+  }
+  return result;
+}
+
+// 4 https://www.codewars.com/kata/514a6336889283a3d2000001
+
+function getEvenNumbers(numbersArray) {
+  return numbersArray.filter((item) => item % 2 === 0);
+}
+
+// 5 https://www.codewars.com/kata/5a090c4e697598d0b9000004
+
+function solve(arr) {
+  let cpyOne = [...arr];
+  let cpyTwo = [...arr];
+
+  cpyOne.sort((a, b) => (a < b ? 1 : -1));
+  cpyTwo.sort((a, b) => (a > b ? 1 : -1));
+
+  let result = [];
+  for (let i = 0; i < cpyOne.length; i++) {
+    if (!result.includes(cpyOne[i])) {
+      result.push(cpyOne[i]);
+    }
+
+    if (!result.includes(cpyTwo[i])) {
+      result.push(cpyTwo[i]);
+    }
+  }
+  return result;
+}
+
+// 6 https://www.codewars.com/kata/566044325f8fddc1c000002c
+
+function evenChars(string) {
+  if (string.length < 2 || string.length > 100) return "invalid string";
+  let result = [];
+
+  for (let i = 1; i < string.length; i += 2) {
+    result.push(string[i]);
+  }
+
+  return result;
+}
+
+// 7 https://www.codewars.com/kata/545a4c5a61aa4c6916000755
+
+function gimme(triplet) {
+  let cpyTriplet = triplet.slice().sort((a, b) => a - b);
+  return triplet.indexOf(cpyTriplet[1]);
+}
+
+// 8 https://www.codewars.com/kata/578553c3a1b8d5c40300037c
+
+function binaryArrayToNumber(arr) {
+  let value = arr.join("");
+  return parseInt(value, 2);
+}
+
+// 9 https://www.codewars.com/kata/585d7d5adb20cf33cb000235
+
+function findUniq(arr) {
+  for (let i = 0; i < arr.length - 1; i++) {
+    if (arr[i] !== arr[i + 1] && arr[i + 1] !== arr[i + 2]) {
```

Comment:
You are returning the same, just changing conditions - it only knots your code and gives no effect.
Besides:
Task warns: The tests contain some very huge arrays, so think about performance.
In such conditions if it is possible - only 1 cycle through huge array and breake the cycle, when you retrieved needed value - save performance.